### PR TITLE
[EDA] Account Owner Update Should Work With the Chosen Administrative Account Record Type

### LINKED ISSUE #1286: [EDA] Account Owner Update Should Work With the Chosen Administrative Account Record Type
Note: FOR INTERNAL USE ONLY. For support questions, please use the EDA Chatter group in The Power of Us Hub.
For feature requests, please use the Ideas tab in The Power of Us Hub.

Do you want to request a feature or report a bug?
bug

What is the current behavior?
The ACCT_IndividualAccounts_TDTM class doesn't respect the record type selected in for 'Administrative Account Record Type' in 'Accounts and Contacts' tab for EDA settings. As a result changing the owner on Contact doesn't update the Account whose record type matches with the record type of 'Administrative Account Record Type'.

What is the expected behavior?
ACCT_IndividualAccounts_TDTM should respect the record type selected in for 'Administrative Account Record Type' in 'Accounts and Contacts' tab for EDA settings so that changing the owner on Contact updates the related Account.

What is the motivation / use case for changing the behavior?
Bug fix

Acceptance Criteria

Given: I am on a EDA Settings page
When: I change the "Default Account Model" and ''Administrative Account Record Type"
Then: The ACCT_IndividualAccounts_TDTM format should respect the record type selected in ''Administrative Account Record Type" selected in EDA Settings.

### PR DESCRIPTION
**Pull request by skosgi1**
# Critical Changes

# Changes
When a Contact Owner is updated for a Contact record, the Account Owner for the Account related to that Contact is now also updated. This change applies to Accounts that have the record type specified in the Administrative Account Record Type setting on the Accounts and Contacts tab in EDA Settings.

This fix applies only to newly created Contacts and doesn't update the owner for existing Accounts.

# Issues Closed
Closes #1286 

# New Metadata

# Deleted Metadata

# Testing Notes
https://salesforce.quip.com/VBbZAI45GTRb#dXCACAhjTK8

### CODE REVIEW

File: src/classes/ACCT_IndividualAccounts_TDTM.cls [hunk 1]
```diff
@@ -29,20 +29,27 @@
...
     */
     private static ID defaultRecTypeId = UTIL_CustomSettingsFacade.getSettings().Account_Processor__c;
 
-    /*******************************************************************************************************
-    * @description The user defined Administrative Account record type.
+    /*****************************************************************************************************************
+    * @description The user defined Administrative Account record type in "ACCOUNTS AND CONTACTS" tab of EDA Settings.
     */
-    private static ID userDefinedAdminRecordTypeId = Util_Describe.getAdminAccRecTypeID();
+    private static ID userDefinedAdminRecordTypeId = Util_Describe.getCustomAdminAccRecTypeId();
```
Comment (skosgi1) [inline src/classes/ACCT_IndividualAccounts_TDTM.cls:50]: This is the only logic change in this class, other changes are to better name variables and spacing fixes.

File: src/classes/ACCT_IndividualAccounts_TEST.cls [hunk 2]
```diff
@@ -2201,4 +2201,47 @@ private class ACCT_IndividualAccounts_TEST {
...
+        insert tempUser;
+
+        Contact con = UTIL_UnitTestData_API.getContact();
+        con.LastName = 'test1234';
+        insert con;
+
+        Test.startTest();
+            con.OwnerId = tempUser.Id;
+            update con;
+        Test.stopTest();
```
Comment (MatthewBlanski) [inline src/classes/ACCT_IndividualAccounts_TEST.cls:2237]: One of the things to note about `Test.startTest()` and `Test.stopTest()` is that their ideal use case is to create a separate set of limits for tests. As such, we should use this only to call methods or use dml operations that we are testing. This allows for isolated profiling when we need to perform it.

```suggestion
        con.OwnerId = tempUser.Id;
        
        Test.startTest();
            update con;
        Test.stopTest();
```

**Comment by skosgi1** [comment]
> omment on a suggestion for specific test blocks that is not required.

Thanks for the suggestion on this one @MatthewBlanski address it on the latest commit.

**Comment by spinkelman** [comment]
QA Passes - Merging
Will close WI

**Pull request merged**